integration/src/main/java/scripts/testscripts/DatasetOperations.java

```diff
@@ -83,8 +84,32 @@ private void setupSnapshot(TestUserSpecification user) throws Exception {
     snapshotId = snapshotsApi.createTestSnapshot(tdrDataset);
   }
 
-  private ObjectNode createMetadata(String name) {
-    return objectMapper.createObjectNode().put("name", name);
+  private ObjectNode createMetadata(
+      String title,
+      String description,
+      String creator,
+      String date,
+      List DataCollections,
+      String url,
+      List generatedBy,
+      List storage,
+      Object counts,
+      List contributors) {
+    var obj =
+        objectMapper
+            .createObjectNode()
```

Comment:
At this point I think it might be easier to create a `Map` here and then use `objectMapper` to convert the `Map` to `String`. Using `Map` would let you create most of the object's content using `Map.of()`.